src/ui/public/saved_objects/saved_object.js

```diff
@@ -0,0 +1,36 @@
+import _ from 'lodash';
+
+export class SavedObject {
+  constructor(client, attributes) {
+    this.client = client;
+    this.attributes = attributes;
+  }
+
+  get(key) {
+    return _.get(this.attributes, key);
+  }
+
+  set(key, value) {
+    return _.set(this.attributes, key, value);
+  }
+
+  save() {
+    if (this.id) {
+      return this.client.update(this.type, this.id, this.attributes);
+    } else {
+      return this.client.create(this.type, this.attributes);
+    }
+  }
+
+  delete() {
+    return this.client.delete(this.type, this.id);
+  }
+
+  get id() {
```

Comment:
There were here as an object always has `id` and `type` though I don't see any real benefit to keep them around.